code-challenges.test.js

```diff
@@ -14,54 +14,139 @@
+    const studyMinutesWeek1 = {
+      sunday: 90,
+      monday: 30,
+      tuesday: 20,
+      wednesday: 15,
+      thursday: 30,
+      friday: 15,
+      saturday: 60
+    }
+    let outPut1 = [15, 15, 20, 30, 30, 60, 90]
+    
+    const studyMinutesWeek2 = {
+      sunday: 100,
+      monday: 10,
+      tuesday: 45,
+      wednesday: 60,
+      thursday: 20,
+      friday: 15,
+      saturday: 65
+    }
+    let outPut2 = [10, 15, 20, 45, 60, 65, 100]
+    
+    expect(smallToBig(studyMinutesWeek1)).toEqual(outPut1)
+    expect(smallToBig(studyMinutesWeek2)).toEqual(outPut2)
+  })
+})
+// test 1 -    ReferenceError: smallToBig is not defined
+// test 2 -   Pass
+
 
 // b) Create the function that makes the test pass.
+  //create function named smallToBig
+  //give function parameter of "object"
+  // within function, return the object passed through the Object.values built in method to return values as an array, and use .sort built in method to organize numbers smallest to largest.
+    // use compare function within .sort to arrange numeric values.
+
+const smallToBig = (object) => {
+  return Object.values(object).sort((a, b) => a - b)
+}
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+describe("arrSum", () => {
+  it("takes in an array and returns an array of the accumulating sum", () => {
+    const accountTransactions1 = [100, -17, -23, -9]
+    let outPut1 = [100, 83, 60, 51]
+    const accountTransactions2 = [250, -89, 100, -96]
+    let outPut2 = [250, 161, 261, 165]
+    const accountTransactions3 = []
+    let outPut3 = []
+
+    expect(arrSum(accountTransactions1)).toEqual(outPut1)
+    expect(arrSum(accountTransactions2)).toEqual(outPut2)
+    expect(arrSum(accountTransactions3)).toEqual(outPut3)
+  })
+})
+// test 1 -    ReferenceError: arrSum is not defined
+// test 2 -    undefined
+// test 3 -    NaN
+// test 4 -    Pass
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
-
-const accountTransactions3 = []
-// Expected output: []
 
 // b) Create the function that makes the test pass.
+  // create function named arrSum
+  // give function parameter of arr 
+  // within function, create for loop starting at index 1 and ceasing iteration at length of arr
+  // within loop, set current index to equal the sum of the current index and the previous index
+  // return arr
+    //NOTE: multiple attempts to utilize .map built in method were made, some kept commented out from some point in their test cycle below. The last one performs the desired function, but was learned through online resources and I don't yet fully understand the syntax.
```

Comment:
Really great attempts to use HOF!  There are several different approaches that can be taken, some of which involve using the reduce method which could be handy to take a look at if you want to dig deeper.